apps/cms/lib/telemetry/reporter.ex

```diff
@@ -0,0 +1,56 @@
+defmodule CMS.Telemetry.Reporter do
+  @moduledoc """
+
+  """
+
+  use GenServer
+
+  require Logger
+
+  alias Telemetry.Metrics
+
+  def start_link(metrics: metrics) do
+    GenServer.start_link(__MODULE__, metrics)
+  end
+
+  @impl true
+  def init(metrics) do
+    Process.flag(:trap_exit, true)
+
+    groups = Enum.group_by(metrics, & &1.event_name)
+
+    for {event, metrics} <- groups do
+      :telemetry.attach({__MODULE__, event, self()}, event, &__MODULE__.handle_event/4, metrics)
```

Comment:
question: do you need a GenServer for this? I don't see it maintaining any state, so the metric source could do the logging directly?

Reply:
afaik, this is how you're supposed to write custom telemetry reporters. it's how the other reporter we're using telemetry_metrics_statsd works: https://github.com/beam-telemetry/telemetry_metrics_statsd/blob/main/lib/telemetry_metrics_statsd.ex. and the console reporter that telemetry metrics ships with: https://github.com/beam-telemetry/telemetry_metrics/blob/main/lib/telemetry_metrics/console_reporter.ex. the idea behind telemetry is to separate out the metric emitters from reporters.